### DISCUSSION TITLE
HTML Output (WIP)

### DISCUSSION THREAD

**Discussion by gateway** [rkhamilton/vqgan-clip-generator#43] [General]
So I finally had some times to write something that generates html from an image folder of upscaled images with meta data.

Personally this is for my use cases where I iterate over the same man text prompt combined with materials and art styles to see which one I like better.

**So far what it does:**

- Create thumbs based upon max width and save as a jpg (smaller file sizes) /thumbs folder
- Create src image based upon max width and save as a jpg /src folder
- Generate a static valid HTML 5 document using Twitters Bootstrap framework
- Added a small lite weight lightbox for image pop ups.

I tried to keep the extra python libs down, I added Path and a sanitize function for file names (names with spaces, or weird chrs in filenames are not valid html)

**Things I would like to do:**

- better styling
- grid view of images
- add document link to a master index file
- Maybe make a web interface to run some commands (prob wont do have time to do this)

I can share my rough code if you like, I'm sure you will laugh at how things are written :)

Output

![image](https://user-images.githubusercontent.com/519563/139561778-b804169d-c515-4911-9d95-b5931b8d95fb.png)

**Comment by rkhamilton** [comment]
Looks like a solid start!

What I find that I want to do when browsing a library like this is try to dial in specific parameters. So, I'll try to arrange my folder view width so that iterations loop on each row. In other words, row 1 will be all the same settings except iterations is increasing from left to right, since iterations has a big effect. Then I can scan down and see the effect of the second thing that's changing.

What I feel like I really want is the ability to dynamically sort by metadata. "Show me all iterations==15, with ascending values of init_weight" for example. I've never coded any kind of UI or interface for anything, so I'm not planning to tackle it, but that's the kind of analysis I want to do.

On a related note, please comment on [my proposal to switch to JPG output here](45).

**Reply by gateway** [reply]
Yea I would like to be able to filter things as well, Ill see what to do but with static web pages with no db or anything and dynamic rebuilding its a bit more work.  I'll look into it.. I just wanted to get a first version out so I can build upon it.

**Reply by gateway** [reply]
Also I would love the ability to do what I think artbreader site did, I think they where using stylegan and had various settings such as genetics they could tweak and then merge or use 2 parents created to create a child image etc.. kinda interesting idea but not sure if its viable.. ideally i like using UI interfaces to build upon my art but im also a command line guru ever since the old days of the Spark workstations..
